src/main/java/io/epirus/console/deploy/DeployRunner.java

```diff
@@ -0,0 +1,158 @@
+import io.epirus.console.account.AccountUtils;
+import io.epirus.console.config.CliConfig;
+import io.epirus.console.project.utils.ProjectUtils;
+import okhttp3.OkHttpClient;
+
+import org.web3j.account.LocalWeb3jAccount;
+import org.web3j.codegen.Console;
+import org.web3j.crypto.Credentials;
+import org.web3j.protocol.Network;
+
+import static io.epirus.console.project.utils.ProjectUtils.uploadSolidityMetadata;
+
+public class DeployRunner {
+    public static final String USAGE = "epirus deploy <network>";
+    private final Path workingDirectory;
+    private final Network network;
+    private final Credentials credentials;
+    private final AccountManager accountManager;
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 1) {
+            new DeployRunner(
+                            Network.valueOf(args[0].toUpperCase()),
+                            new AccountManager(
+                                    CliConfig.getConfig(
+                                            CliConfig.getDefaultEpirusConfigPath().toFile()),
+                                    new OkHttpClient()))
+                    .deploy();
+        } else {
+            Console.exitError(USAGE);
+        }
+    }
+
+    public DeployRunner(Network network, AccountManager accountManager, Path workingDirectory) {
+        this.workingDirectory = workingDirectory;
+        this.network = network;
+        Path walletPath = ProjectUtils.loadProjectWalletFile(workingDirectory).get();
+        Path walletPasswordPath =
+                ProjectUtils.loadProjectPasswordWalletFile(workingDirectory).get();
+        this.credentials = ProjectUtils.createCredentials(walletPath, walletPasswordPath);
+        this.accountManager = accountManager;
+    }
+
+    public DeployRunner(Network network, AccountManager accountManager) {
+        this.workingDirectory = Paths.get(System.getProperty("user.dir"));
+        this.network = network;
+        Path walletPath = ProjectUtils.loadProjectWalletFile(workingDirectory).get();
+        Path walletPasswordPath =
+                ProjectUtils.loadProjectPasswordWalletFile(workingDirectory).get();
+        this.credentials = ProjectUtils.createCredentials(walletPath, walletPasswordPath);
+        this.accountManager = accountManager;
+    }
+
+    public void deploy() throws Exception {
+        AccountUtils.accountInit(accountManager);
+        this.accountManager.checkIfAccountIsConfirmed();
+        fundWallet();
+        waitForBalanceUpdate();
+        uploadSolidityMetadata(network, workingDirectory);
+        runGradle(workingDirectory);
+    }
+
+    private void fundWallet() {
+        int tries = 5;
+        while (tries-- > 0) {
+            try {
+                BigInteger accountBalance = accountManager.getAccountBalance(credentials, network);
+                if (accountBalance.equals(BigInteger.ZERO)) {
+                    WalletFunder.fundWallet(
```

Comment:
I don't understand why we need a retry loop for fundWallet?

Reply:
Alexandrou has had errors with the faucet (due to weird on-chain events lately), so I think that's where this comes from.

Reply:
loop removed if. If getting the account balance fails or funding the wallet fails it will do Console.exitError with the exception message.